packages/server-wallet/src/protocols/__test__/open-channel.test.ts

```diff
@@ -58,7 +59,12 @@ describe(`funding phase`, () => {
       await crankAndAssert(obj, {fundsToDeposit: 5});
       // and then doesn't submit it a second time
       await crankAndAssert(obj, {fundsToDeposit: 0});
-    });
+      // but then let's "wait" for 10 minutes for the fund request to the chain service to get stale
+      await ChainServiceRequest.query(knex)
+        .findOne({channelId: testChan.channelId})
+        .patch({timestamp: new Date(Date.now() - requestTimeout - 1)});
+      await crankAndAssert(obj, {fundsToDeposit: 5});
```

Comment:
Might be worth adding a test to ensure that it doesn't retry after 10mins if the transaction went through?

Reply:
You are actually pointing out something that potentially could be an issue. The chain service request retry does not depend on the success/failure of the chain service request. At the moment, every chain service request is associated with a chain event: the funding request is associated with the deposit event; the close + withdraw request is associated with the asset transferred event.

So the retry will happen in the following case:
1. 10 minutes have elapsed.
1. The protocol is rerun.
1. The chain service request has **succeeded**.
1. The wallet has not received a chain event.

But maybe that's ok for now?

Reply:
I think it's probably ok for the receipt of the chain event to be the signal that the chain service request succeeded, which is essentially the situation you're describing above.

The thing it might be worth confirming is that if a chain event _has_ been received, then no retry occurs. I expect that to work, but it would be nice to be sure.

Reply:
Thanks for the feedback! If all prefund states are singed (a prerequisite for a fund chain service request), then a chain service requests will get submitted if:
1. There is not a valid existing chain service request. This is currently tested here https://github.com/statechannels/statechannels/blob/chain-service-resilience/packages/server-wallet/src/protocols/__test__/open-channel.test.ts#L57.
2. The channel needs the participant's funds. This is already tested with existing unit tests. One of which is here: https://github.com/statechannels/statechannels/blob/chain-service-resilience/packages/server-wallet/src/protocols/__test__/open-channel.test.ts#L81 

We should add a test similar to 1 for the close protocol.

Reply:
Tests added in ed415705d.